app-services/secrets/README.md

```diff
@@ -1,36 +1,116 @@
 # Secrets example
 
-#### Overview
-
 This example demonstrates storing a secret to the secret store (Vault) and retrieving those secrets.
 
-When running an application service in secure mode, secrets can be stored by making an HTTP `POST` call to `http://[host]:[port]/api/v1/secrets`.  If running in insecure mode, secrets can be configured in consul or in the config yaml file.
+If in secure mode, the secrets are stored and retrieved from Vault based on the *SecretStoreExclusive* section of the configuration file.
+
+Please refer to the [Application Functions SDK documentation](https://github.com/edgexfoundry/app-functions-sdk-go#secrets)  for more details on storing and getting secrets using the SDK.
+
+## Run the app within EdgeX docker
+
+**Steps:**
+
+1. From the app-service-examples root folder, build the docker image for the secrets example service.
+
+   ```
+   make docker_secrets_example
+   ```
+
+2. Add the environment variable `ADD_SECRETSTORE_TOKENS` to the vault-worker service in your EdgeX docker compose file. It takes a list of service keys that it generates Vault tokens for. This is used as the base path for the service's secrets in Vault. In the example below, `edgex-appservice` is the service key for the shared database credentials for Application Services.  `secrets-example` is the application service that we will be running.
+
+   ```yml
+   environment:
+         - "SECRETSTORE_SETUP_DONE_FLAG=/tmp/edgex/secrets/edgex-consul/.secretstore-setup-done"
+         - "ADD_SECRETSTORE_TOKENS=edgex-appservice, secrets-example"
+       
+   ```
+
+3. The default configuration is for running natively and we use environment overrides in the compose files for use in docker.
+
+4. Add the secrets example application service to your EdgeX docker compose file. This includes the environment overrides for the SecretStore and SecretStoreExclusive configurations:
+
+   ```yml
+     app-service-secrets-example:
+       image: edgexfoundry/docker-secrets-example:dev
+       container_name: app-service-secrets-example
+       hostname: app-service-secrets-example
+       ports:
+         - "48095:48095"
+       networks:
+         - edgex-network
+       environment:
+         <<: *common-variables
+         SecretStore_Protocol: https
+         SecretStore_TokenFile: /tmp/edgex/secrets/edgex-appservice/secrets-token.json
+         SecretStoreExclusive_Host: edgex-vault
+         SecretStoreExclusive_Protocol: https
```

Comment:
This should be the default, so shouldn't need override. If not fix in your config toml.